File: static/js/datetime.js
// Name : datetime.js

// Get time (24h format)
var curTime=function() {
    var now = new Date();
    var hours = now.getHours();
    var minutes = now.getMinutes();
    if (hours < 10) {hours = "0" + hours;}
    if (minutes < 10) {minutes = "0" + minutes;}
    var time = hours + "h" + minutes;
    $("#heure").html(time);

};

// Get date (dayname/daynum/month format)
// Translate current month and current dayname to fr
var curDate=function() {
    var dicMonth = new Array("Janv.", "Fev.", "Mars", "Avr.", "Mai", "Juin",
                        "Juil.", "Août", "Sept.", "Oct.", "Nov.", "Déc.");
    var dicDay = new Array("Dim.", "Lun.", "Mar.", "Mer.", "Jeu.", "Ven.", "Sam.");
    var now = new Date();
    // Current day (numeric format)
    var dayNum = now.getUTCDate();
    // Current day of the week (0-6)
    var day = now.getDay();
    var dayFr = dicDay[day];
    // Current month (0-11)
    var month = now.getUTCMonth();
    var monthFr = dicMonth[month];
    //if (day < 10) {day = "0" + day;}
    $("#date").html(dayFr + " " + dayNum + " " + monthFr);
};


// Display time and date when page is loaded
// Auto refresh time every 60 seconds
$(document).ready(function() {
    curTime();
    curDate();
    setInterval(curTime, 60000);
    });

File: src/datetime.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
#
# Name : datetime.py
# Get current time and date, translate date to French

import time
import os

# Setup timezone
os.environ['TZ'] = 'Europe/Paris'
time.tzset()

# Request time (24h format)
def getTime():
    cur_time=time.strftime('%Hh%M')
    return cur_time

# Request date and translate it
def getDate():
    cur_day=time.strftime('%A')
    cur_dayn=time.strftime('%d')
    cur_month=time.strftime('%B')

    dico_jour = {'Monday':'Lun.',
            'Tuesday':'Mar.',
            'Wednesday':'Mer.',
            'Thursday':'Jeu.',
            'Friday':'Ven.',
            'Saturday':'Sam.',
            'Sunday':'Dim.'}
    for word, replace in dico_jour.items():
        cur_day=cur_day.replace(word, replace)

    dico_mois = {'January':'Janv.',
            'February':'Fev.',
            'March':'Mars',
            'April':'Avr.',
            'May':'Mai',
            'June':'Juin',
            'July':'Juil.',
            'August':'Aout',
            'September':'Sept.',
            'October':'Oct.',
            'November':'Nov.',
            'December':'Dec.'}
    for word, replace in dico_mois.items():
        cur_month=cur_month.replace(word, replace)

    return (cur_day+' '+cur_dayn+' '+cur_month)
